### ISSUE TITLE
muted (/mute) outbound messages excluded from irc_raw (/server raw)

### ISSUE THREAD

**Issue by mumixam** [weechat/weechat#980]
When /mute is used to send a message to the server, the message is not shown in irc_raw (/server raw) buffer. The message should be shown in order to debug scripts or triggers that may use /mute

```/mute /msg -server local #test  test```

**Comment by flashcode** [comment]
This is the normal behavior : `/mute` without arguments mutes *ALL* buffers, so even the IRC raw buffer if it's open. If you close and reopen this raw buffer, the message should be displayed.
One solution would be to mute only the buffer where you don't want output, instead of all.

As I said on IRC, one solution would be to not mute the IRC raw buffer by default (with a buffer property or local variable saying it should not be muted).
And a new argument can be added in command `/mute` to force the mute everywhere if needed (so really all buffers).

**Comment by mumixam** [comment]
here is my workaround that statically exempts irc_raw from being muted
```
diff --git a/src/gui/gui-chat.c b/src/gui/gui-chat.c
index 5fc018d..f938b63 100644
--- a/src/gui/gui-chat.c
+++ b/src/gui/gui-chat.c
@@ -673,11 +673,13 @@ gui_chat_printf_date_tags (struct t_gui_buffer *buffer, time_t date,
     }
 
     /* if mute is enabled for buffer (or all buffers), then just return */
-    if ((gui_chat_mute == GUI_CHAT_MUTE_ALL_BUFFERS)
-        || ((gui_chat_mute == GUI_CHAT_MUTE_BUFFER)
-            && (gui_chat_mute_buffer == buffer)))
-        return;
-
+    if (string_strcasecmp (gui_buffer_get_string( buffer, "name" ), "irc_raw") != 0)
+    {
+        if ((gui_chat_mute == GUI_CHAT_MUTE_ALL_BUFFERS)
+            || ((gui_chat_mute == GUI_CHAT_MUTE_BUFFER)
+                && (gui_chat_mute_buffer == buffer)))
+            return;
+    }
     weechat_va_format (message);
     if (!vbuffer)
         return;
```

**Comment by flashcode** [comment]
@mumixam: yes this works but can not be a solution to the problem. By design, weechat core is completely separated from plugins and it should not do anythiing related to a specific plugin loaded.
So as I said I'll add a local variable or buffer property to not mute some buffers, and the irc plugin itself will enabled that on its raw buffer.
